troi/musicbrainz/year_lookup.py

```diff
@@ -0,0 +1,63 @@
+import re
+import sys
+import uuid
+from urllib.parse import quote
+
+import requests
+import ujson
+
+from troi import Element, Artist, Recording, PipelineError
+
+class YearLookupElement(Element):
+    '''
+        Look up musicbrainz earliest release year for a list of recordings, based on artist credit name and recording name.
+    '''
+
+    SERVER_URL = "http://bono.metabrainz.org:8000/acrp-year-lookup/json"
+
+    def __init__(self):
+        Element.__init__(self)
+
+    @staticmethod
+    def inputs():
+        return [ Recording ]
+
+    @staticmethod
+    def outputs():
+        return [ Recording ]
+
+    def read(self, inputs, debug=False):
+
+        recordings = inputs[0]
+        if not recordings:
+            return []
+
+        data = []
+        for r in recordings:
+            data.append({ '[recording_name]': r.name, 
+                          '[artist_credit_name]': r.artist.name })
+
+        r = requests.post(self.SERVER_URL, json=data)
+        if r.status_code != 200:
+            raise PipelineError("Cannot fetch recording years from MusicBrainz: HTTP code %d" % r.status_code)
+
+        try:
+            rows = ujson.loads(r.text)
+        except ValueError as err:
+            raise PipelineError("Cannot fetch recording years from MusicBrainz: " + str(err))
+
+        mbid_index = {}
+        for row in rows:
+            mbid_index[row['artist_credit_name'] + row['recording_name']] = row['year']
```

Comment:
If you pass in more than 1 AC + recording, and they all fold into a single one, the AC and recording name in the result will only include the last instance of those values. 

e.g. 
AC: Queen,queen; recording: Death on two legs,death-on-two-legs
only returns keys queen, death-on-two-legs

Reply:
Fixed in bono-data-sets.